pkg/sdk/testint/tables_integration_test.go

```diff
@@ -0,0 +1,949 @@
+		request := sdk.NewCreateTableRequest(id, columns).
+			WithOutOfLineConstraint(outOfLineConstraint).
+			WithStageFileFormat([]sdk.StageFileFormatRequest{*stageFileFormat}).
+			WithStageCopyOptions([]sdk.StageCopyOptionsRequest{*stageCopyOptions}).
+			WithComment(&comment).
+			WithDataRetentionTimeInDays(sdk.Int(30)).
+			WithMaxDataExtensionTimeInDays(sdk.Int(30))
+
+		err := client.Tables.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		table, err := client.Tables.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assertTable(t, table, id)
+		assert.Equal(t, table.Comment, comment)
+		assert.Equal(t, 30, table.RetentionTime)
+
+		param, err := client.Parameters.ShowObjectParameter(ctx, sdk.ObjectParameterMaxDataExtensionTimeInDays, sdk.Object{ObjectType: sdk.ObjectTypeTable, Name: table.ID()})
+		assert.NoError(t, err)
+		assert.Equal(t, "30", param.Value)
+
+		tableColumns := getTableColumnsFor(t, client, table.ID())
+		expectedColumns := []expectedColumn{
+			{"COLUMN_3", sdk.DataTypeVARCHAR},
+			{"COLUMN_1", sdk.DataTypeVARCHAR},
+			{"COLUMN_2", sdk.DataTypeNumber},
+		}
+		assertColumns(t, expectedColumns, tableColumns)
+	})
+
+	t.Run("create table as select", func(t *testing.T) {
+		maskingPolicy, _ := createMaskingPolicyWithOptions(t, client, database, schema, []sdk.TableColumnSignature{
+			{
+				Name: "col1",
+				Type: sdk.DataTypeVARCHAR,
+			},
+		}, sdk.DataTypeVARCHAR, "REPLACE('X', 1)", nil)
+		columns := []sdk.TableAsSelectColumnRequest{
+			*sdk.NewTableAsSelectColumnRequest("COLUMN_3").
+				WithType_(sdk.Pointer(sdk.DataTypeVARCHAR)).
+				WithCopyGrants(sdk.Bool(true)).
+				WithOrReplace(sdk.Bool(true)),
+			*sdk.NewTableAsSelectColumnRequest("COLUMN_1").
+				WithType_(sdk.Pointer(sdk.DataTypeVARCHAR)).
+				WithCopyGrants(sdk.Bool(true)).
+				WithOrReplace(sdk.Bool(true)),
+			*sdk.NewTableAsSelectColumnRequest("COLUMN_2").
+				WithType_(sdk.Pointer(sdk.DataTypeVARCHAR)).
+				WithCopyGrants(sdk.Bool(true)).
+				WithOrReplace(sdk.Bool(true)).WithMaskingPolicyName(sdk.Pointer(sdk.NewSchemaObjectIdentifier(database.Name, schema.Name, maskingPolicy.Name))),
+		}
+
+		name := random.String()
+		id := sdk.NewSchemaObjectIdentifier(database.Name, schema.Name, name)
+		request := sdk.NewCreateTableAsSelectRequest(id, columns)
+
+		err := client.Tables.CreateAsSelect(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		table, err := client.Tables.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		tableColumns := getTableColumnsFor(t, client, table.ID())
+		expectedColumns := []expectedColumn{
+			{"COLUMN_3", sdk.DataTypeVARCHAR},
+			{"COLUMN_1", sdk.DataTypeVARCHAR},
+			{"COLUMN_2", sdk.DataTypeVARCHAR},
+		}
+		assertColumns(t, expectedColumns, tableColumns)
+	})
+
+	// TODO: fix this test, it should create two integer column but is creating 3 text ones instead
```

Comment:
next pr, in the future, or just forgot to come back to it in this pr ?

Reply:
future